source/simulator/core.cc

```diff
@@ -1969,18 +1969,58 @@ namespace aspect
               }
           }
 
+        // get new time step size
+        TimeStepping::Reaction reaction;
+        const double new_time_step = time_stepping_manager.compute_time_step_size(reaction);
 
+        switch (reaction)
+          {
+            case TimeStepping::Reaction::refine_and_repeat_step:
+            {
+              pcout << "Refining the mesh based on the time stepping manager ..." << std::endl;
+              refine_mesh(max_refinement_level);
 
-        // if we postprocess nonlinear iterations, this function is called within
-        // solve_timestep () in the individual solver schemes
-        if (!parameters.run_postprocessors_on_nonlinear_iterations)
-          postprocess ();
+              // fall-through into the next switch case
+              DEAL_II_FALLTHROUGH;
+            }
+            case TimeStepping::Reaction::repeat_step:
+            {
+              pcout << "Repeating the current time step based on the time stepping manager ..." << std::endl;
```

Comment:
The wording here and below is a bit confusing to me. We repeat the `time step number`, but with a different `time step size` (so earlier in time). Yet the parameter is called `new_time_step`, with which is meant the time step size.